Frends.Community.Multipart.SendMultipartRequest/SendMultipartRequest.cs

```diff
@@ -1,53 +1,86 @@
+                foreach (var file in input.FilePaths)
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+
+                    if (!File.Exists(file.Fullpath)) 
+                        throw new FileNotFoundException("Input file was not found. File: " + file.Fullpath);
+
+                    var filebyte = File.ReadAllBytes(file.Fullpath);
+                    request.AddFile("file", filebyte, file.Name, "application/octet-stream");
+                }
+
+                foreach (var file in input.TextData)
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+                    request.AddParameter(file.Key, file.Value, ParameterType.GetOrPost);
+                }
 
-            foreach (var header in input.Headers) if (header.Name != "Content-Type") request.AddHeader(header.Name, header.Value);
+                foreach (var header in input.Headers) 
+                    if (header.Name != "Content-Type") 
+                        request.AddHeader(header.Name, header.Value);
 
-            request.AddHeader("Content-Type", "multipart/form-data");
-            if (options.Authentication is AuthenticationMethod.Basic) client.Authenticator = new HttpBasicAuthenticator(options.Username, options.Password);
-            else if (options.Authentication is AuthenticationMethod.OAuth2) client.Authenticator = new OAuth2AuthorizationRequestHeaderAuthenticator("Bearer", options.BearerToken);
-            var response = await client.ExecuteAsync(request);
+                request.AddHeader("Content-Type", "multipart/form-data");
 
-            return new SendResult { Body = JsonConvert.DeserializeObject<dynamic>(response.Content), RequestIsSuccessful = response.IsSuccessful, ErrorException = response.ErrorException, ErrorMessage = response.ErrorMessage };
-        }
-    }
-}
+                if (options.Authentication is AuthenticationMethod.Basic) 
+                    client.Authenticator = new HttpBasicAuthenticator(options.Username, options.Password);
+                else if (options.Authentication is AuthenticationMethod.OAuth2) 
+                    client.Authenticator = new OAuth2AuthorizationRequestHeaderAuthenticator(options.BearerToken, "Bearer");
+
+                var response = await client.ExecuteAsync(request, cancellationToken);
+
+                if (!response.IsSuccessful) 
+                    throw new WebException($"{response.StatusCode}: {response.StatusDescription}");
+
+                return new SendResult { Body = response.Content != null ? JsonConvert.DeserializeObject<dynamic>(response.Content) : null, RequestIsSuccessful = response.IsSuccessful, ErrorException = response.ErrorException, ErrorMessage = response.ErrorMessage };
+
+            }
+            catch (Exception ex)
+            {
+                if (options.HandleErrors)
+                    throw new Exception(ex.ToString());
+                else
+                    throw;

```

Comment:
What's the idea in this catch-block? Both of these options basically do the same thing. If the try-block goes to an error, this catch-block would throw the same error which would be thrown without `try-catch`.

Reply:
Had to create some extra exception logs of our test process and forgot to remove for prod. Removed now.